portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,23 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+//handles the preloading -> https://stackoverflow.com/questions/25253391/javascript-loading-screen-while-page-loads
+function onReady(callback) {
+  var intervalId = window.setInterval(function() {
+    if (document.getElementsByTagName('body')[0] !== undefined) {
+      window.clearInterval(intervalId);
+      callback.call(this);
+    }
+  }, 3000);
+}
+
+function setVisible(selector, visible, type) {
+  document.querySelector(selector).style.display = visible ? type : 'none';
+}
+
+onReady(function() {
+    setVisible('#header', true, 'flex');
```

Comment:
Instead of setting these styles in javascript (which crosses your View and Controller), you should create some sort of 'hidden' css class that has something like `display: none`. You can then toggle that class directly with `element.toggleClass('hidden', false)`.